src/solarized8.vim

```diff
@@ -466,6 +466,17 @@ for s:solarized_background in ["dark", "light"]
     call s:put("hi! ColorColumn"    .s:fmt_none   .s:fg_none   .s:bg_base02)
     call s:put("hi! Cursor"         .s:fmt_none   .s:fg_base3  .s:bg_blue)
     call s:put("hi! link lCursor Cursor")
+
+    call s:put("if exists(':terminal')")
```

Comment:
I thought it made sense to check for the `:terminal` capability, rather than looking at the version of the software. Much like in web dev, where [browser detection is bad](https://css-tricks.com/browser-detection-is-bad/) and capability testing is encouraged. Hypothetically, if at a future date Vim added support for `:terminal`, then these styles could **just work** ™️ in Vim as well. Of course, it's possible that Vim would use `hl-CursorTerm` instead of `hl-TermCursor` in which case we'd need to add more `if/else` branches anyway...